src/tzdb.rs

```diff
@@ -344,29 +346,80 @@ fn resolve_posix_tz_string_for_epoch_seconds(
         });
     };
 
-    let start = &dst_variant.start_date;
-    let end = &dst_variant.end_date;
-
-    // TODO: Resolve safety issue around utils.
-    //   Using f64 is a hold over from early implementation days and should
-    //   be moved away from.
-
-    let (is_transition_day, transition) =
-        cmp_seconds_to_transitions(&start.day, &end.day, seconds)?;
-
-    let transition =
-        compute_tz_for_epoch_seconds(is_transition_day, transition, seconds, dst_variant);
     let std_offset = UtcOffsetSeconds::from(&posix_tz_string.std_info);
     let dst_offset = UtcOffsetSeconds::from(&dst_variant.variant_info);
-    let (old_offset, new_offset, transition) = match transition {
-        TransitionType::Dst => (std_offset, dst_offset, start),
-        TransitionType::Std => (dst_offset, std_offset, end),
-    };
+
     let year = utils::epoch_time_to_epoch_year(seconds * 1000);
-    let year_epoch = utils::epoch_days_for_year(year) * 86400;
-    let leap_day = utils::mathematical_in_leap_year(seconds * 1000) as u16;
+    let dst_start_seconds =
+        calculate_transition_seconds_for_year(year, dst_variant.start_date, std_offset);
+    let dst_end_seconds =
+        calculate_transition_seconds_for_year(year, dst_variant.end_date, dst_offset);
+
+    // Need to determine if the range being tested is standard or savings time.
+    let dst_is_inversed = dst_end_seconds < dst_start_seconds;
+
+    // We have potentially to different variations of the DST start and end time.
+    //
+    // Northern hemisphere: dst_start -> dst_end
```

Comment:
I assume this also handles Ireland's backwards DST

Reply:
Short answer: I'm not sure

Longer answer:

That's the one caveat to this current approach that will need to be checked.

Currently, it's not too much of an issue though because most of our source data is all in rearguard. So I didn't want to add handling for the vanguard format without a good way to test it for now.